light/client.go

```diff
@@ -268,16 +264,16 @@ func (c *Client) restoreTrustedLightBlock() error {
 	if err != nil {
 		return ErrGetTrustedBlockHeight{Err: err}
 	}
-
-	if lastHeight > 0 {
-		trustedBlock, err := c.trustedStore.LightBlock(lastHeight)
-		if err != nil {
-			return ErrGetTrustedBlock{Err: err}
-		}
-		c.latestTrustedBlock = trustedBlock
-		c.logger.Info("Restored trusted light block", "height", lastHeight)
+	if lastHeight == -1 {
+		return ErrEmptyTrustedStore
 	}
 
+	trustedBlock, err := c.trustedStore.LightBlock(lastHeight)
```

Comment:
Did you test using `lastHeight == 0`? This situation is covered in the original version.